src/engine/readaheadmanager.cpp

```diff
@@ -150,10 +177,13 @@ void ReadAheadManager::hintReader(double dRate, HintVector* pHintList) {
     // cache.
     SINT length_to_cache = 2 * CachingReaderChunk::kSamples;
 
+    // this called after the precious chunk was consumed
+    int sample = SampleUtil::roundPlayPosToFrameStart(
+            m_currentPosition, kNumChannels);
     current_position.length = length_to_cache;
     current_position.sample = in_reverse ?
-            m_iCurrentPosition - length_to_cache :
-            m_iCurrentPosition;
+            sample - length_to_cache :
```

Comment:
And here again, the first sample depends on the direction. I didn't notice before, so please revert the API of CachingReader back to the old way where the value of first sample already depends on the direction.